benchmark/src/benchmark/core.clj

```diff
@@ -1,20 +1,79 @@
 (ns benchmark.core
-  (:require [benchmark.measure :as m]
+  (:require [clojure.tools.cli :as cli]
+            [benchmark.measure :as m]
             [benchmark.config :as c]
-            [clojure.pprint :as pp]))
-
-
-(defn -main [& _]
-  (let [measurements (vec (for [config       c/db-configs
-                                initial-size c/initial-datoms
-                                n            c/datom-counts
-                                _            (range c/iterations)]
-                              (m/measure-performance-full initial-size n config)))]
-    (->> measurements
-         (apply concat)
-         (group-by :context)
-         (mapv (fn [[context group]]
-                 {:context context
-                  :mean-time (/ (reduce (fn [x y] (+ x (:time y))) 0 group)
-                                (count group))}))
-         pp/pprint)))
+            [benchmark.store :refer [transact-missing-schema transact-data ->RemoteDB]]
+            [clojure.pprint :refer [pprint]]))
+
+
+(def cli-options
+  [["-u" "--db-server-url URL" "Base URL for datahike server, e.g. http://localhost:3000"
+    :default nil]
+   ["-n" "--db-name DBNAME" "Database name for datahike server" :default nil]
+   ["-g" "--db-token TOKEN" "Token for datahike server" :default nil]
+   ["-t" "--tag TAG" "Add tag to measurements"
+    :default #{}
```

Comment:
So far, the tag parameter is not designed to be unique, so you can add as many tags as you want to. My thought was that this would give us the maximum freedom. So for now we can use a tag to identify the branch the benchmarks are coming from but maybe at some point in the future more information would be helpful to specify specific benchmarks.

With multiple values the default is used as initial value to be updated with the assoc-fn or update-fn, so having an empty set is simply convenient to have as default. I just discovered a default-fn function that could be used instead, so I can check if that works as I see that empty-set as tag default looks weird. If the update-fn works I can also exchange that for the assoc-fn because that would make it look also less complicated I'd say.

At the moment, I am checking if the server description is complete by checking for nil values. If all of the values needed are non-nil, the result is sent to the server, a partial description is detected early and prints a message about that and ends the program, and if all of the values are nil, the cli simply prints the results to stdout. With that, people can still run and save the benchmarks on their local machine without the need to have to send them to some server. 
Do you think, we should change the defaults anyway and just decide depending on if a token is given or not?

Reply:
I think it is good to have it default to print to stdout and the way you check for complete/partial/nothing is a good approach. The tags are fine with this behavior as well. I just thought it looks better if you don't show these default values here.